File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Repository/PacienteRepository.java
package com.example.BackendCitasMedicas.Repository;


import com.example.BackendCitasMedicas.Model.Paciente;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface PacienteRepository extends JpaRepository<Paciente, Long> {
}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Repository/PacienteRepository.kt
package com.example.frontcitasmedicas.Repository


import com.example.frontcitasmedicas.Model.Paciente
import com.example.frontcitasmedicas.Service.RetrofitClient

class PacienteRepository {

    private val api = RetrofitClient.pacienteService

    suspend fun getPacientes() = api.getAllPacientes()

    suspend fun getPaciente(id: Long) = api.getPacienteById(id)

    suspend fun addPaciente(paciente: Paciente) = api.addPaciente(paciente)

    suspend fun updatePaciente(paciente: Paciente) = api.updatePaciente(paciente)

    suspend fun deletePaciente(id: Long) = api.deletePaciente(id)
}
